File: node/ssl-certificates-chain-api-sample.js
let https = require('https');
let querystring = require('querystring');

const apiUrl = 'https://api.threatintelligenceplatform.com/v1/sslCertificatesChain';

const parameters = {
    apiKey: 'Your SSL Certificates Chain API key',
    domainName: 'threatintelligenceplatform.com'
};

let url = apiUrl + '?' + querystring.stringify(parameters);

https.get(url, function (res) {
    let rawData = '';

    res.on('data', function(chunk) {
        rawData += chunk;
    });

    res.on('end', function () {
        try {
            let parsedData = JSON.parse(rawData);

            if (parsedData.constructor.name === 'Array')
                console.log(
                    'Chain length: ' + parsedData.length);
            else
                console.dir(parsedData, {colors: true});
        }
        catch (e) {
            console.log(e.message);
        }
    });
}).on('error', function(e) {
    console.log('Error: ' + e.message);
});

File: python/ssl-certificates-chain-api-sample.py
try:
    from urllib.request import urlopen, pathname2url
except ImportError:
    from urllib import pathname2url
    from urllib2 import urlopen

import json


def print_response(txt):
    response_json = json.loads(txt)
    print('Chain length: ' + str(len(response_json)))


api_key = 'Your SSL Certificates Chain API key'
domain = 'threatintelligenceplatform.com'

url = 'https://api.threatintelligenceplatform.com/v1/sslCertificatesChain'\
    + '?apiKey=' + pathname2url(api_key)\
    + '&domainName=' + pathname2url(domain)

try:
    response = urlopen(url).read().decode('utf8')
    print_response(urlopen(url).read().decode('utf8'))
except Exception as e:
    print(e)
